authorizer.py

```diff
@@ -85,6 +87,9 @@ def _VerifyServingSession(self, institution, session):
     if (serving_session.login_type == "schedule" and
         self.handler.request.path == "/spots_available"):
```

Comment:
It might be nice to make a set of a few different pages available for each login_type:
```python
allowed_paths = {
  "schedule": set(["/spots_available", "/catalog_print", "/welcome"]),
  "preregistration": set(["/catalog_print", "/welcome"])}
if allowed_paths[serving_session.login_type] == self.handler.request.path:
  return True
# Note: We could put the main paths (below) into the above map too.
if "/" + serving_session.login_type == self.handler.request.path:
  return True
logging.error("Access not allowed for /%s for user with serving session %s" % (
  self.handler.request.path, serving_session.login_type))
return False
```